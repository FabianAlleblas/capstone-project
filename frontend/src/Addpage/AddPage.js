import styled from 'styled-components/macro'
import AddPlantForm from '../Components/Forms/AddPlantForm'
import Header from '../Components/Header/Header'

export default function AddPage({ savePlantData }) {
  return (
    <PageWrapper>
      <FixedHeader secondaryStyle>Add Plant</FixedHeader>
      <AddPlantForm savePlantData={savePlantData} />
    </PageWrapper>
  )
}

const FixedHeader = styled(Header)`
  left: 0;
  position: fixed;
  top: 0;
`

const PageWrapper = styled.div`
  background-color: var(--primary-dark);
  display: flex;
  height: 100vh;
  place-content: center;
  align-items: center;
`
