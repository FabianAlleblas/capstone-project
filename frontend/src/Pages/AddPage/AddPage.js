import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import AddPlantForm from '../../Components/Forms/AddPlantForm'
import Header from '../../Components/Header/Header'

AddPage.propTypes = {
  savePlantData: PropTypes.func.isRequired,
}

export default function AddPage({ savePlantData }) {
  return (
    <>
      <FixedHeader secondaryStyle>Add Plant</FixedHeader>
      <FormContainer>
        <AddPlantForm savePlantData={savePlantData} />
      </FormContainer>
    </>
  )
}

const FixedHeader = styled(Header)`
  left: 0;
  position: fixed;
  top: 0;
`

const FormContainer = styled.main`
  align-items: center;
  background-color: var(--primary-dark);
  display: flex;
  height: 100vh;
  padding: 40px 40px 0;
  place-content: center;
`
