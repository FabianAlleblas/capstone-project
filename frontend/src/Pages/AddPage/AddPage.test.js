import { render } from '@testing-library/react'
import 'jest-styled-components'
import AddPage from './AddPage'

const savePlantDataMock = jest.fn()

describe('AddPage', () => {
  it('renders correctly', () => {
    const { container } = render(<AddPage savePlantData={savePlantDataMock} />)
    expect(container).toMatchSnapshot()
  })
})
