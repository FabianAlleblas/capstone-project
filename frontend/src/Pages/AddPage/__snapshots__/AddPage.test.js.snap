// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPage renders correctly 1`] = `
.c11 {
  background-color: var(--primary-light);
  border-radius: 22px;
  border: none;
  box-shadow: 0 0 4px var(--shadow);
  color: var(--primary-dark);
  font-family: 'Josefin sans';
  font-size: 1.25rem;
  font-weight: 700;
  padding: 12px 16px 8px;
  width: 100%;
}

.c13 {
  background-color: var(--primary-dark);
  border-radius: 22px;
  border: 4px solid var(--primary-light);
  box-shadow: 0 0 4px var(--shadow);
  color: var(--primary-light);
  font-family: 'Josefin sans';
  font-size: 1.25rem;
  font-weight: 700;
  padding: 8px 12px 4px;
  width: 100%;
}

.c5 {
  background-color: var(--primary-dark);
  display: grid;
  gap: 20px;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-image: url();
  background-position: center;
  background-size: cover,contain;
  border-radius: 50%;
  border: 4px dashed var(--primary-light);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 148px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto 10px;
  position: relative;
  width: 148px;
}

.c6 svg {
  margin: 0 4px 4px 0;
}

.c7 {
  height: 0.1px;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  width: 0.1px;
  z-index: -1;
}

.c8 {
  color: var(--form-font-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 1.2rem;
}

.c9 {
  border-radius: 22px;
  border: none;
  color: var(--form-font-color);
  display: block;
  font-family: 'Josefin sans',sans-serif;
  font-size: 1rem;
  margin-top: 8px;
  padding: 14px 20px 10px;
  width: 100%;
}

.c9:focus {
  box-shadow: 0 0 4px var(--shadow) inset;
  outline: none;
}

.c10 {
  display: grid;
  gap: 20px;
  grid-template-columns: 1fr 1fr;
  padding: 30px 0 0;
}

.c12 {
  opacity: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  z-index: 10;
}

.c2 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  background: var(--primary-dark);
  border-radius: 0 0 50px 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 20px 30px 13px 20px;
}

.c3 {
  color: var(--primary-light);
  font-size: 1.75rem;
  font-weight: 700;
  line-height: 1;
}

.c1 {
  left: 0;
  position: fixed;
  top: 0;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--primary-dark);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  padding: 40px 40px 0;
  place-content: center;
}

<div>
  <header
    class="c0 c1"
  >
    <div
      class="c2"
      data-testid="heading-container"
    >
      <h1
        class="c3"
      >
        Add Plant
      </h1>
    </div>
  </header>
  <main
    class="c4"
  >
    <form
      class="c5"
    >
      <label
        class="c6"
        data-testid="img-input-wrapper"
      >
        <input
          accept="image/png, image/jpeg"
          alt="image-input"
          class="c7"
          name="image"
          type="file"
        />
        <svg>
          addimgicon.svg
        </svg>
      </label>
      <label
        class="c8"
      >
        Your plants name*:
        <input
          class="c9"
          maxlength="28"
          name="name"
          placeholder="Bob"
          required=""
          type="text"
        />
      </label>
      <label
        class="c8"
      >
        The species of your plant*:
        <input
          class="c9"
          maxlength="28"
          name="species"
          placeholder="Monstera deliciosa"
          required=""
          type="text"
        />
      </label>
      <label
        class="c8"
      >
        Special infos:
        <input
          class="c9"
          maxlength="28"
          name="specialInfo"
          placeholder="variegated"
          type="text"
        />
      </label>
      <div
        class="c10"
      >
        <button
          class="c11 c12"
        >
          Add Plant
        </button>
        <button
          class="c13"
          type="button"
        >
          Cancel
        </button>
      </div>
    </form>
  </main>
</div>
`;
